catalyst-gateway/bin/src/cardano/event.rs

```diff
@@ -0,0 +1,56 @@
+//! Event types for Chain Indexer.
+
+use cardano_blockchain_types::Slot;
+
+/// Represents various events that can occur in the chain indexer.
+/// These events are used to track the state and progress of indexing operations.
+#[derive(Debug, Clone)]
+pub(crate) enum ChainIndexerEvent {
+    /// Event triggered when the synchronization process starts.
+    SyncStarted,
+    /// Event triggered when the synchronization process completes.
+    SyncCompleted,
+    /// Event triggered when the number of current synchronization tasks changes.
+    SyncTasksChanged {
+        /// The current number of synchronization tasks.
+        current_sync_tasks: u16,
+    },
+    /// Event triggered when the live tip slot changes.
+    LiveTipSlotChanged {
+        /// The new live tip slot.
+        slot: Slot,
+    },
+    /// Event triggered when the immutable tip slot changes.
+    ImmutableTipSlotChanged {
+        /// The new immutable tip slot.
+        slot: Slot,
+    },
+    /// Event triggered when the indexed slot progresses.
+    IndexedSlotProgressed {
+        /// The latest indexed slot.
+        slot: Slot,
+    },
+    /// Event triggered when live data is purged.
+    LiveDataPurged {
```

Comment:
Lets make two different purging events, Forward purging (which should be triggered for `cardano_chain_follower::Kind::Rollback`, purging itself for this case its not yet implemented) and Backwards purging which is done for this case
```
            // IF there is only 1 chain follower left in sync_tasks, then all
            // immutable followers have finished.
            // When this happens we need to purge the live index of any records that exist
            // before the current immutable tip.
            // Note: to prevent a data race when multiple nodes are syncing, we probably
            // want to put a gap in this, so that there are X slots of overlap
            // between the live chain and immutable chain.  This gap should be
            // a parameter.
   ```